content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,794 @@
+type the value will be and allow it to perform type specific optimiztions
+thus potentially making resultant code faster:
+[source,clojure]
+----
+user=> (def ^Integer five 5)
+#'user/five
+user=> (meta #'five)
+{:ns #<Namespace user>, :name five, :column 1, :line 1, :file "NO_SOURCE_PATH", :tag java.lang.Integer}
+----
+We can see in that example the `:tag` property is set.
+
+You can also stak the shorthand notations:
+[source,clojure]
+----
+user=> (def ^Integer ^:debug ^:private five 5)
+#'user/five
+user=> (meta #'five)
+{:ns #<Namespace user>, :name five, :column 1, :private true, :debug true, :line 1, :file "NO_SOURCE_PATH", :tag java.lang.Integer}
+----
+
+* <<xref/../../reference/metadata#,Clojure Official Documentation>>
+* http://en.wikibooks.org/wiki/Learning_Clojure/Meta_Data[Learning Clojure: Meta Data]
+
+== ' - Quote macro
+
+Can be used against symbols as part of a dispatch macro
+(see <<xref/../weird_characters#dispatch,`#`>>). Also used to quote forms
+and prevent their evalutation as with the quote function.
+[source,clojure]
+----
+user=> (1 3 4) ; fails as it tries to evaluate 1 as a function
+
+ClassCastException java.lang.Long cannot be cast to clojure.lang.IFn  user/eval925 (NO_SOURCE_FILE:1)
+user=> '(1 3 4) ; quote
+(1 3 4)
+user=> (quote (1 2 3)) ; using the longer quote method
+(1 2 3)
+user=>
+----
+
+* <<xref/../../reference/special_forms#quote,Clojure Official Documentation>>
+
+== ; - Comment
+
+`;` is a comment. In fact it's a comment *macro* that takes all input from its
+starting point to the end of the line and ensures that the reader ignores it.
+[source,clojure]
+----
+user=> (def x "x") ; this is a comment
+#'user/x
+user=> ; this is a comment too
+<returns nothing>
+----
+
+== : - Keyword
+
+`:` is the indicator for a keyword which is an interned string that provides
+fast comparison and lower memory overhead.
+
+[source,clojure]
+----
+user=> (type :test)
+clojure.lang.Keyword
+----
+Alternatively you can use `keyword` to create a keyword from a string
+[source,clojure]
+----
+user=> (keyword "test")
+:test
+----
+A neat thing about keywords is that they alsoe implement `IFn` and can act as
+functions for extracting values from maps which is very nice:
+[source,clojure]
+----
+user=> (def my-map { :one 1 :two 2 })
+#'user/my-map
+user=> (:one my-map) ; get the value for :one by invoking it as function
+1
+user=> (:three my-map) ; it can safely access non-keys
+nil
+user=> (:three my-map 3) ; it can return a default if specified
+3
+----
+
+* <<xref/../../reference/data_structures#Keywords,Clojure Official Documentation>>
+
+[[autoresolved_keys]]
+== +::+ - Autoresolved keyword
+
+`::` is used to autoresolve a keyword to the current namespace:
+[source,clojure]
+----
+user=> :my-keyword
+:my-keyword
+user=> ::my-keyword
+:user/my-keyword
+user=> (= ::my-keyword :my-keyword)
+false
+----
+This is useful when creating macros. If you want to ensure a macro, that calls
```

Comment:
I don't think I understand the sentence starting with "If you want". Maybe drop this remark?